### ISSUE TITLE
Implement support for polygons with holes

### ISSUE THREAD

**Issue by astrofrog** [astropy/regions#2]
(like shapely does)

**Comment by astrofrog** [comment]
@cdeil - I wonder if we could say that PixelPolygonRegion should be a single region and this could be implemented with composite regions?

**Comment by cdeil** [comment]
@astrofrog - Can we close one of #2 or #3 to focus discussion in one place?

And I don't have an opinion on this question at this point.
Just getting *any* polygons implemented would be fine with me as a first step.
Also, looking to ds9 to see what they support / do might be a good idea ... it's seen decades of use and we can just do what they did (if not too hard to implement).

**Comment by astrofrog** [comment]
I agree on closing one (or both) of them since I think those use cases are covered by compositing.

I'm spending 1 hour now to try and see how hard getting polygon support in is.

**Comment by cdeil** [comment]
> I agree on closing one (or both) of them since I think those use cases are covered by compositing.

I suggest to keep one open.
At the very least we need a docs section how to achieve this.

> I'm spending 1 hour now to try and see how hard getting polygon support in is.

Thanks!
I suggest you start by looking at Ginga (see link in #69).
Probably we can just copy over the function there for pixel polygon.
(and adapt API / add tests / docs ... it's still a day of work)
If you put the code, I might try to find time to add tests / docs later this week.
